### ISSUE TITLE
Minor setup issues

### ISSUE THREAD

**Issue by jleavers** [cloudfoundry/diego-windows-release#12]
1) setup.ps1

When I first ran this it errored as follows:

```
Import-LocalizedData : Cannot find the Windows PowerShell data file 'PSDesiredStateConfiguration.Resource.psd1' in
directory 'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\PSDesiredStateConfiguration\en\', or in any parent
culture directories.
At C:\Windows\system32\WindowsPowerShell\v1.0\Modules\PSDesiredStateConfiguration\PSDesiredStateConfiguration.psm1:57
```

When I checked, I didn't have an en directory - only en-US - so I had to copy en-US -> en.

2) generate.exe

This errored as follows:

```
C:\cf>generate.exe -outputDir=c:\cf -windowsUsername=user -windowsPassword=pass -boshUrl=https://admin:pass@x.x.x.x:25555
panic: Invalid boolean: '' at line 786, column 16

goroutine 1 [running]:
main.FailOnError(0x29db40, 0xc0820c25b0)
        C:/diego-windows-release/greenhouse-install-script-generator/src/generat
e/generate.go:306 +0x57
main.main()
        C:/diego-windows-release/greenhouse-install-script-generator/src/generat
e/generate.go:118 +0xbe5

goroutine 8 [runnable]:
net/http.(*persistConn).readLoop(0xc08208c0b0)
        c:/go/src/net/http/transport.go:928 +0x9d5
created by net/http.(*Transport).dialConn
        c:/go/src/net/http/transport.go:660 +0xca6

goroutine 9 [select]:
net/http.(*persistConn).writeLoop(0xc08208c0b0)
        c:/go/src/net/http/transport.go:945 +0x424
created by net/http.(*Transport).dialConn
        c:/go/src/net/http/transport.go:661 +0xcc3
```
        
I checked the existing Diego BOSH manifest - line 786 was properties.consul.require_ssl. This had been left as null, so I changed it to true to resolve the problem.

As SSL is in use, perhaps it should have been set to true anyway, but arguably as the default is true (https://github.com/cloudfoundry-incubator/consul-release/blob/0ad37ffa52115a572f91430ee8875c521c4f8575/jobs/consul_agent/spec) could this also be assumed by generate.exe?

3) CF dependency

With cf-release you might check the commit SHA at https://github.com/cloudfoundry-incubator/diego-cf-compatibility/blob/master/compatibility-v2.csv to ensure you used a compatible diego, garden-linux, & etcd release.

Are there any similar dependencies on the parent cf release when using diego-windows-release & garden-windows-release? 

Or do the two Windows installs just have to match, e.g. diego-windows v0.309 & garden-windows v0.97.

Just wondering whether it's likely there will be future versions that have specific cf dependencies.

**Comment by mhoran** [comment]
Hey @jleavers,

> 1) setup.ps1

> When I checked, I didn't have an en directory - only en-US - so I had to copy en-US -> en.

We're not sure why this is. What are the region settings on your server? We just set up a new Windows 2012R2 VM and selected the United Kingdom region and language, and could not reproduce this issue. There's nothing about our script that should be locale specific, so I'm not sure why you're seeing this. However, we have had other reports of this as well. Unfortunately we've been unable to reproduce, and don't know why our script would fail to run in another locale.

> 2) generate.exe

> As SSL is in use, perhaps it should have been set to true anyway, but arguably as the default is true (https://github.com/cloudfoundry-incubator/consul-release/blob/0ad37ffa52115a572f91430ee8875c521c4f8575/jobs/consul_agent/spec) could this also be assumed by generate.exe?

Unfortunately there's not much we can do about this one. The YAML parsing library we use, [candiedyaml](https://github.com/cloudfoundry-incubator/candiedyaml) does not handle empty elements well. A null value should be fine, but it seems your manifest contains the empty string, which cannot be parsed as a boolean.

> 3) CF dependency

> With cf-release you might check the commit SHA at https://github.com/cloudfoundry-incubator/diego-cf-compatibility/blob/master/compatibility-v2.csv to ensure you used a compatible diego, garden-linux, & etcd release.

> Are there any similar dependencies on the parent cf release when using diego-windows-release & garden-windows-release?

So long as the Diego release you've chosen is compatible with the CF release you've deployed, as specified by the compatibility matrix, you should be fine. We constantly deploy the latest `release-candidate` CF branch, which runs against the latest `master` Diego. This is exactly what the Diego team does when generating the compatibility matrix.

**Comment by jleavers** [comment]
Hi,

1)

The server Locale is set to en-GB:

```
C:\Windows\system32>systeminfo | find "Locale"
System Locale:             en-gb;English (United Kingdom)
Input Locale:              en-gb;English (United Kingdom)
```
However, Powershell thinks it is en-US:

```
PS C:\Users\Administrator> echo $PsUICulture
en-US
```
I have just checked a fresh Server 2012 install and C:\Windows\System32\WindowsPowerShell\v1.0\Modules\PSDesiredStateConfiguration only contains en-US.

However, if the variable above is en-US you would think it would work fine...

2)

To clarify, the value was 'null' - not an empty string:

```
properties:
  consul:
    agent:
      <snip>
      require_ssl: null
```

3)

Thanks, when I glanced at https://github.com/cloudfoundry/garden-windows-release/releases it listed specific Diego commits as compatible, so I wanted to make sure.

However all seems fine so far, with cf v230 / diego v0.1450.0 / diego-windows v0.309 / garden-windows v0.97.
